src/containers/array.c

```diff
@@ -144,7 +144,7 @@ static inline int32_t grow_capacity(int32_t capacity) {
     return (capacity <= 0) ? ARRAY_DEFAULT_INIT_SIZE
                            : capacity < 64 ? capacity * 2
                                            : capacity < 1024 ? capacity * 3 / 2
-                                                             : capacity * 5 / 4;
+                                                             : (capacity & 0xfffff000) + 4096;
```

Comment:
It it customary to grow by a fixed factor beyond a certain point. Growing by a fixed amount can bring a significant penalty at scale, at least in theory.

It is not unreasonable, however, to grow by units of 4kB for large bitsets, but the number of units should grow as the bitset grows.